### PR TITLE
Add missed checks on methods of the closed connection

### PR DESCRIPTION
**Pull request by nicktorwald**
Add missed checks on methods of the closed connection.

Closes: #72
Affects: #119, #74

### CODE REVIEW

File: src/main/java/org/tarantool/jdbc/SQLConnection.java [hunk 1]
```diff
@@ -171,25 +177,38 @@ public String nativeSQL(String sql) throws SQLException {
...
     }
 
     @Override
     public void commit() throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        checkNotClosed();
+        if (getAutoCommit()) {
+            throw new SQLNonTransientException(
+                    "Cannot commit when auto-commit is enabled.",
+                    SQLStates.IVALID_TRANSACTION_STATE.getSqlState()
```
Comment (Totktonada) [inline src/main/java/org/tarantool/jdbc/SQLConnection.java]: Typo: IVALID → INVALID.

File: src/main/java/org/tarantool/jdbc/SQLConnection.java [hunk 2]
```diff
@@ -171,25 +177,38 @@ public String nativeSQL(String sql) throws SQLException {
...
+            );
+        }
```
Comment (Totktonada) [inline src/main/java/org/tarantool/jdbc/SQLConnection.java]: Please, add raising SQLFeatureNotSupportedException() or just TODO / XXX mark. Pass though here looks confusing for me. The same for rollback().

File: src/main/java/org/tarantool/jdbc/SQLConnection.java [hunk 3]
```diff
@@ -296,21 +324,43 @@ public int getHoldability() throws SQLException {
 
     @Override
     public Savepoint setSavepoint() throws SQLException {
+        checkNotClosed();
+        if (getAutoCommit()) {
+            throw new SQLNonTransientException(
+                    "Cannot set a savepoint when auto-commit is enabled.",
```
Comment (Totktonada) [inline src/main/java/org/tarantool/jdbc/SQLConnection.java]: > if setSavepoint is invoked outside of an active transaction, a transaction will be started at this newly created savepoint.

https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#setSavepoint--

The same for setSavepoint(name).

File: src/test/java/org/tarantool/jdbc/JdbcConnectionIT.java [hunk 4]
```diff
@@ -186,10 +180,95 @@ public void testPrepareHoldableStatement() throws SQLException {
...
-            conn.close();
-            conn.prepareStatement(sqlString, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY, ResultSet.HOLD_CURSORS_OVER_COMMIT);
-        });
+    }
+
+    @Test
+    public void testUnavailableMethodsAfterClose() throws SQLException {
+        conn.close();
+
+        assertThrows(SQLException.class, () -> conn.clearWarnings());
```
Comment (Totktonada) [inline src/test/java/org/tarantool/jdbc/JdbcConnectionIT.java]: Maybe it worth to check SQLSTATE here?

File: src/main/java/org/tarantool/jdbc/SQLConnection.java (hunk 3 above)
Comment (nicktorwald) [reply src/main/java/org/tarantool/jdbc/SQLConnection.java]: Can you clarify your comment a bit?
This highlighted block follows another spec requirement: 
SQLException - if a database access error occurs, this method is called while participating in a distributed transaction, this method is called on a closed connection or **this Connection object is currently in auto-commit mode**

Comment (Totktonada) [reply src/main/java/org/tarantool/jdbc/SQLConnection.java]: I understood now: setSavepoint() cannot be called in auto-commit mode, but when the mode is disabled it is not necessary to start a transaction explicitly somehow.

**Pull request merged**